state/factoidstate.go

```diff
@@ -436,6 +436,18 @@ func (fs *FactoidState) GetCoinbaseTransaction(dbheight uint32, ftime interfaces
 				m[v] = struct{}{}
 			}
 
+			totalGrantAmount := uint64(0)
+			for i, o := range desc.Outputs {
+				if _, ok := m[uint32(i)]; !ok {
+					totalGrantAmount += o.GetAmount()
+				}
+			}
+
+			// Auto-cancel when the total amount exceeds the grant limit
+			if totalGrantAmount > constants.COINBASE_PAYOUT_GRANT_LIMIT {
+				return coinbase
```

Comment:
Ignore grant payouts altogether when total amount exceeds the grant limit

Reply:
Is the limit supposed to be set on a grant round by grant round basis?

So when we hardcode in the next round of grants, would we also update `COINBASE_PAYOUT_GRANT_LIMIT` if there is a new limit?

Reply:
As long as these limits are internal I don't think there is coming back from it after these versions have been deployed. Followers running an older version which do not know about a higher COINBASE_PAYOUT_GRANT_LIMIT will not save these blocks to the database until they are upgraded.
Maybe it's a better idea is to go back to the drawing board and allow the limits to be set in an adminblock entry just like ANO efficiency values are altered. But I can't determine how big the impact will be for nodes running older versions and if a change like that is desirable as it's effectively a protocol change.

Reply:
Good point. I think adding admin block changes is a bit overkill, we do have the ability to cancel assuming we catch it in time.

I think some discussion around this would be beneficial to determine what the limit is.
Best case scenario is we calculate the amount in the grant pool, and use that as the limit, rather than a hard coded ceiling.